spec/features/admin/configuration/tax_rates_spec.rb

```diff
@@ -29,4 +29,37 @@
     click_button "Create"
     expect(page).to have_content("Tax rate has been successfully created!")
   end
+
+  # Adds further CRUD operations: editing, deleting
+  context "while editing" do
+    it "fields can be filled in and dropfdowns retains changes" do
+      visit spree.edit_admin_tax_rate_path(tax_rate.id)
+      fill_in "Rate", with: "0.23"
+      fill_in "Name", with: "GST"
+
+      find(:id, "tax_rate_zone_id").select "Ilhas"
+      find(:id, "tax_rate_tax_category_id").select "Full"
+      click_button "Update"
+      expect(page).to have_content('Tax rate "GST" has been successfully updated!')
+      expect(page).to have_content("0.23")
+    end
+
+    # See #6554: in order to set a Tax Rate as included in the price,
+    # there must be at least one Zone set the "Default Tax Zone"
+    it "checkboxes can be ticked" do
+      visit spree.edit_admin_tax_rate_path(tax_rate.id)
+      uncheck("tax_rate[show_rate_in_label]")
+      check("tax_rate[included_in_price]")
+      click_button "Update"
+      expect(page).to have_content("Included Price Validation")
+    end
+
+    it "can be deleted", js: true do
+      click_link "Tax Rates"
+      accept_alert do
+        find(:xpath, "/html/body/div[1]/div[3]/div/div/div/table/tbody/tr[1]/td[8]/a[2]").click
```

Comment:
This is pretty hard to read and understand (for anyone that has to come and change it in the future) and it's probably a bit fragile as well; it could break easily if the layout of the page changes at all, like for example if there's a slight change to the layout on admin pages, this spec could start failing.

We could do this, and it would delete the correct tax rate if there were multiple tax rates:
```ruby
find("#spree_tax_rate_#{tax_rate.id} .delete-resource").click
```
But in this test there's only ever one tax rate present, so we could even just do this:
```ruby
find(".delete-resource").click
```

Either one looks good to me. Much easier to read, and hopefully less prone to breaking if the layout changes slightly :ok_hand:

Reply:
This did look somewhat bulky/fragile and not readable. Thanks for suggesting this alternative!